.github/update_model_info.py

```diff
@@ -10,117 +10,124 @@
 # limitations under the License.
 
 import argparse
-import hashlib
-import json
 import os
 import shutil
 import tempfile
-import zipfile
 
-from monai.utils import look_up_option
+from utils import (
+    compress_bundle,
+    download_large_files,
+    get_changed_bundle_list,
+    get_checksum,
+    get_hash_func,
+    get_json_dict,
+    push_new_model_info_branch,
+    save_model_info,
+    upload_bundle,
+)
 
-SUPPORTED_HASH_TYPES = {"md5": hashlib.md5, "sha1": hashlib.sha1, "sha256": hashlib.sha256, "sha512": hashlib.sha512}
-
-
-def get_sub_folders(root_dir: str):
-    return [f.name for f in os.scandir(root_dir) if f.is_dir()]
-
-
-def get_json_dict(json_dict_path: str):
-    with open(json_dict_path, "r") as f:
-        json_dict = json.load(f)
-    return json_dict
-
-
-def save_model_info(model_info_dict, model_info_path: str):
-    with open(model_info_path, "w") as f:
-        json.dump(model_info_dict, f)
 
+def update_model_info(bundle_name: str, models_path: str = "models", model_info_file: str = "model_info.json"):
+    """
+    For a changed model (bundle), this function is used to do the following steps in order to update it:
+
+    1. check if bundle folder exists.
+    2. create a temporary copy of the bundle.
+    3. download large files (if having the corresponding config file) into the copy.
+    4. compress the copy.
+    5. upload a compressed copy.
+    6. update `model_info_file`.
+
+    Returns:
+        a 2-tuple.
+        If update successful, the form is (True,"").
+        If update failed, the form is (False, "error reason")
+    """
 
-def get_hash_func(hash_type: str = "sha1"):
-    actual_hash_func = look_up_option(hash_type.lower(), SUPPORTED_HASH_TYPES)
-    return actual_hash_func()
+    # step 1
+    bundle_path = os.path.join(models_path, bundle_name)
+    if not os.path.exists(bundle_path):
+        return (False, "Bundle path not exist")
 
+    # temp_dir = "test_tmp"
+    # if not os.path.exists(temp_dir):
+    #     os.makedirs(temp_dir)
 
-def update_model_info(models_path: str, model_info_file: str = "model_info.json"):
+    # step 2
+    temp_dir = tempfile.mkdtemp()
+    temp_path = os.path.join(temp_dir, bundle_name)
```

Comment:
You could put every line in this function after this one within one large `try/finally` block with `shutil.rmtree(temp_dir)` as the only line in the finally block. This would ensure that the directory is cleaned up no matter where the exception happens and means you only need the rmtree line once.